osquery/core/windows/process.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <sstream>
+#include <vector>
+
+#include <signal.h>
+
+#include <sys/types.h>
+#include <boost/algorithm/string.hpp>
+
+#include "osquery/core/process.h"
+
+namespace {
+osquery::PlatformPidType __declspec(nothrow) duplicateHandle(osquery::PlatformPidType src) {
+  osquery::PlatformPidType handle = osquery::kInvalidPid;
+
+  if (src != osquery::kInvalidPid) {
+    if (!::DuplicateHandle(GetCurrentProcess(),
+                           src,
+                           GetCurrentProcess(),
+                           &handle,
+                           0,
+                           FALSE,
+                           DUPLICATE_SAME_ACCESS)) {
+      handle = osquery::kInvalidPid;
+    }
+  }
+  return handle;
+}
+}
+
+namespace osquery {
+
+PlatformProcess::PlatformProcess(PlatformPidType id) {
+  id_ = duplicateHandle(id);
+}
+
+PlatformProcess::PlatformProcess(PlatformProcess&& src) {
+  id_ = kInvalidPid;
+  std::swap(id_, src.id_);
+}  
+
+PlatformProcess::~PlatformProcess() { 
+  if (id_ != kInvalidPid) {
+    ::CloseHandle(id_);
+    id_ = kInvalidPid;
+  }
+}
+
+bool PlatformProcess::operator==(const PlatformProcess& process) const {
+  return (::GetProcessId(nativeHandle()) == ::GetProcessId(process.nativeHandle()));
+}
+
+bool PlatformProcess::operator!=(const PlatformProcess& process) const {
+  return (::GetProcessId(nativeHandle()) != ::GetProcessId(process.nativeHandle()));
+}
+
+int PlatformProcess::pid() const {
+  return (int) ::GetProcessId(id_);
+}
+
+bool PlatformProcess::kill() const {
+  if (id_ == kInvalidPid) {
+    return false;
+  }
+  
+  return (::TerminateProcess(id_, 0) != FALSE);
+}
+
+std::shared_ptr<PlatformProcess> PlatformProcess::launchWorker(const std::string& exec_path, const std::string& name) {
```

Comment:
Is it possible to get github to run some kind of clang-format pass? Would solve a ton of these issues automatically.